packages/core/src/schema/common.ts

```diff
@@ -10,6 +23,7 @@ export type ScalarColumn<
   " scalar": scalar;
   " optional": optional;
   " list": list;
+  " default": DefaultValue<scalar>;
```

Comment:
I think `" default"` may need to be implemented similar to the other properties in this type. The type needs to be as strict as possible so that when we go to infer the types for the store methods (`db.create()`, ...), they are also strictly typed.

This is going to be tricky because of the interplay between, scalar, optional, and default, but I think something closer to this would work:

```ts
export type ScalarColumn<
  scalar extends Scalar = Scalar,
  optional extends boolean = boolean,
  list extends boolean = boolean,
  _default extends optional extends false
    ? scalar | undefined
    : scalar | null | undefined = optional extends false
    ? scalar | undefined
    : scalar | null | undefined,
> = {
  " type": "scalar";
  " scalar": scalar;
  " optional": optional;
  " list": list;
  " default": _default;
};
```

This also applies to the other column types in this file.